apis/apps/v1alpha1/cluster_types.go

```diff
@@ -600,6 +608,31 @@ type ClusterNetwork struct {
 	PubliclyAccessible bool `json:"publiclyAccessible,omitempty"`
 }
 
+type ServiceRef struct {
+	// name of the predefined service reference. references the serviceRefDeclaration name defined in clusterDefinition.componentDefs[*].serviceRefDeclarations[*].name
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// When referencing a service provided by other KubeBlocks cluster, you need to provide the name of the Cluster being referenced. and the referenced cluster should be in the same namespace.
+	// By default, when other KubeBlocks Cluster are referenced, the ClusterDefinition.spec.connectionCredential secret corresponding to the referenced Cluster will be used to bind to the current component.
+	// If you require an additional connection credential secret, please set it using the ConnectionCredential.
+	// If you have specified a Cluster, please do not specify a ServiceDescriptor.
+	// +optional
+	Cluster string `json:"cluster,omitempty"`
+
+	// serviceDescriptor defines the service descriptor of the service provided by external sources.
+	// When referencing a service provided by external sources, you need to provide the ServiceDescriptor object name to establish the service binding. The ServiceDescriptor object should be in the same namespace with the cluster.
```

Comment:
It's weird and difficult to maintain for users who has a namespace-based multi-tenant design to require the descriptor object to be in the same namespace.